### PR TITLE
[MRG] more docs update for 4.0 - minor cleanup and additions

### PR DESCRIPTION
**Pull request by ctb**
Minor documentation fixes and updates from a complete docs review.

Note:
- [x] check the migration docs link on the index page specifically, since that's been broken so much!

## Checklist

- [ ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
ready for review & (assuming tests pass) merge, @luizirber and @bluegenes !

**Comment by ctb** [comment]
(I'd like to get this in soon, because it fixes the link to the migration docs from the front page of the docs!)

File: README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ The latest major release is sourmash v4, which has several
 command-line and Python incompatibilities with previous
 versions. Please
 [visit our migration guide](https://sourmash.readthedocs.io/en/latest/support.html#migrating-from-sourmash-v3-x-to-sourmash-4-x)
-to upgrade!
+to ugprade!
```
Comment (luizirber) [inline README.md]:
```suggestion
to upgrade!
```

**Review by luizirber — APPROVED** [review]

**Review by bluegenes — APPROVED** [review]
lgtm!

File: README.md (hunk 1 above)
Comment (ctb) [reply README.md]: argh I think my mind just skips over this change again and again :)

**Pull request merged**